### PR TITLE
Simple Payments

### PR DESCRIPTION
**Pull request by artpi**
This introduces Simple Payments feature.

This is an "ecommerce light" feature.

# Stark team: we dont merge this. The Jetpack release lead does. Dont merge!

zrzut ekranu 2017-07-24 o 19 30 44

More details in projectpaymentbutton-p2

Warning: on calypso side, there still is work to do, but we are pretty certain this JP piece is production-ready.
Since there is no UI in JP, users will start using it when we relese it in calypso.

We are still tweaking CSS, otherwise we consider it ready

## Testing

1. Check this branch in your JP sandbox
1. Build calypso in dev
1. Go to post/page editor
1. Click insert -> payment button
1. Fill product details (email has to be a user email on PP sandbox infrastructure. You can use `artur.piszek+test2@automattic.com`)
1. This inserts the shortcode
1. Go to your page frontend
1. Do `localStorage.setItem('simple-payments-env', 'sandbox' );` to switch to Sandbox mode, refresh
1. Click "pay with paypal" to go with the flow. You can use my test sandbox account (`artur.piszek+test@automattic.com` , pass: `potato123`)

### CODE REVIEW

**Comment by kraftbj** [comment]
Marking it as a blocker as the Stark team has requested this make it into 5.2, so we need to review and decide before code freeze. The Stark team all willing to address feedback to help make it in on time if @artpi is in his offline hours.

File: modules/simple-payments/paypal-express-checkout.js [hunk 1]
```diff
@@ -15,7 +15,12 @@ var PaypalExpressCheckout = {
 	wpRestAPIHost: 'https://public-api.wordpress.com',
 	wpRestAPIVersion: '/wpcom/v2',
 
-	sandbox: true,
+	getEnvironment: function() {
```
Comment (retrofox) [inline modules/simple-payments/paypal-express-checkout.js]: 👍

File: modules/simple-payments/simple-payments.css [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (kraftbj) [inline modules/simple-payments/simple-payments.css]: Should we add this to the concatenation in class.jetpack.php? That's a non-blocker for merge, but should be done or issued so we can get it in there.

File: modules/simple-payments/simple-payments.php [hunk 3]
```diff
@@ -0,0 +1,254 @@
+<?php
+/*
+ * Simple Payments lets users embed a PayPal button fully integrated with wpcom to sell products on the site.
+ * This is not a proper module yet, because not all the pieces are in place. Until everything is shipped, it can be turned
+ * into module that can be enabled/disabled.
+ * TODO: Once the feature is fully shipped, create a file modules/simple-payments.php with a proper header to turn module on/off.
```
Comment (zinigor) [inline modules/simple-payments/simple-payments.php]: After a conversation with @rralian it turned out that there is no plan to make this into a module. The way it's done now will be enough to have this code always available. Keep in mind though that users will still be able to restrict loading these files using filters.

File: modules/simple-payments/simple-payments.php [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+			'cta'         => get_post_meta( $product->ID, 'spay_cta', true ),
+			'multiple'    => get_post_meta( $product->ID, 'spay_multiple', true ) || '0'
+		), $attrs );
+		$data['price'] = $this->format_price(
+			get_post_meta( $product->ID, 'spay_price', true ),
+			get_post_meta( $product->ID, 'spay_currency', true ),
+			$data
+		);
+		$data['id'] = $attrs['id'];
+		if ( ! wp_script_is( 'paypal-express-checkout','enqueued' ) ) {
```
Comment (zinigor) [inline modules/simple-payments/simple-payments.php]: Minor: missing space after comma.

File: modules/simple-payments/simple-payments.php [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+			wp_enqueue_script( 'paypal-express-checkout' );
+		}
+
+		wp_add_inline_script( 'paypal-express-checkout', "try{PaypalExpressCheckout.renderButton( '{$data['blog_id']}', '{$attrs['id']}', '{$data['dom_id']}', '{$data['multiple']}' );}catch(e){}" );
```
Comment (zinigor) [inline modules/simple-payments/simple-payments.php]: Paranoid mode: do you think it would be better to use `sprintf` and `esc_js` for all variables here?

**Review by zinigor — CHANGES_REQUESTED** [review]
I have added several minor comments, will actually test the functionality tomorrow so we can make it until code freeze.

File: modules/simple-payments/simple-payments.php (hunk 3 above)
Comment (artpi) [inline modules/simple-payments/simple-payments.php]: Removed this comment

File: modules/simple-payments/simple-payments.php (hunk 4 above)
Comment (artpi) [inline modules/simple-payments/simple-payments.php]: Done

File: modules/simple-payments/simple-payments.php (hunk 5 above)
Comment (artpi) [inline modules/simple-payments/simple-payments.php]: Done

**Review by zinigor — APPROVED** [review]
Tested, works well, changes look good, thanks!

File: modules/simple-payments/simple-payments.php [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+	private function register_scripts() {
+		/**
+		 * Paypal heavily discourages putting that script in your own server:
+		 * @see https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/add-paypal-button/
+		 */
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js', array(), null, true );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ),
+			array( 'jquery', 'paypal-checkout-js' ), self::$version );
+		wp_enqueue_style( 'simple-payments', plugins_url( '/simple-payments.css', __FILE__ ) );
```
Comment (georgestephanis) [inline modules/simple-payments/simple-payments.php]: Why is this enqueueing, rather than just registering?

**Pull request merged**